python/dgl/distributed/graph_partition_book.py

```diff
@@ -664,14 +700,29 @@ def __init__(self, part_id, num_parts, node_map, edge_map, ntypes, etypes):
         self._num_partitions = num_parts
         self._ntypes = [None] * len(ntypes)
         self._etypes = [None] * len(etypes)
+        self._canonical_etypes = [None] * len(etypes)
+        # map etypes to canonical ones
+        self._etype2canonical = {}
         for ntype in ntypes:
             ntype_id = ntypes[ntype]
             self._ntypes[ntype_id] = ntype
         assert all(ntype is not None for ntype in self._ntypes), \
                 "The node types have invalid IDs."
-        for etype in etypes:
-            etype_id = etypes[etype]
-            self._etypes[etype_id] = etype
+        for etype, etype_id in etypes.items():
+            if isinstance(etype, tuple):
+                assert len(etype) == 3, \
+                    'Canonical etype should be in format of (str, str, str).'
+                c_etype = etype
+                etype = etype[1]
+                self._etypes[etype_id] = etype
+                self._canonical_etypes[etype_id] = c_etype
+                if etype in self._etype2canonical:
+                    self._etype2canonical[etype] = tuple()
+                else:
+                    self._etype2canonical[etype] = c_etype
+            else:
```

Comment:
Could you throw a deprecation warning here?

Reply:
added.